gvfs-helper.c

```diff
@@ -2820,10 +2820,13 @@ static void do_req(const char *url_base,
 		 * with the main Git server (because the cache-server
 		 * doesn't handle 401s).
 		 *
-		 * TODO Think about if we really need to handle this case.
-		 * TODO Guard with "if (params->sever_type == __MAIN)"
+		 * Set an empty u/p to get CURL to automatically negotiate
+		 * for us.  This is necessary to get NTLM negotiation to work.
+		 * TODO We might want to only do this for the __MAIN server
+		 * (because the cache-server doesn't support it).
 		 */
 		curl_easy_setopt(slot->curl, CURLOPT_HTTPAUTH, CURLAUTH_ANY);
+		curl_easy_setopt(slot->curl, CURLOPT_USERPWD, ":");
```

Comment:
We'll want to verify that this doesn't make the cache server OR cloud situation break. I wonder if there is anything we can do to tell if we are against the on-prem server or not.

An installer is sent to the customer, which should help.

Reply:
Yeah, however, the cache server never sends back 401s,
only a generic `400 Bad Request` (with a human-readable
"creds required" message in the html body (sigh)).  So the
cache server will never trigger the 401 logic inside libcurl.
So I think this change is net-neutral WRT the cache servers.

But now that you say that, I am tempted to move the cache
server case into its own branch (and kill the TODO comment).

A larger take-away here is that cache servers (that require
auth) won't work with origin servers that use NTLM.

I'll fix this up and submit another version shortly.